src/System.Net.Sockets.Legacy/tests/FunctionalTests/ArgumentValidationTests.cs

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    // TODO:
+    //
+    // End*:
+    // - Invalid asyncresult type
+    // - asyncresult from different object
+    // - asyncresult with end already called
+    public class ArgumentValidation
+    {
+        private readonly static byte[] s_buffer = new byte[1];
+        private readonly static IList<ArraySegment<byte>> s_buffers = new List<ArraySegment<byte>> { new ArraySegment<byte>(s_buffer) };
+        private readonly static Socket s_ipv4Socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+        private readonly static Socket s_ipv6Socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
+
+        private static void TheAsyncCallback(IAsyncResult ar)
+        {
+        }
+
+        private static Socket GetSocket(AddressFamily addressFamily = AddressFamily.InterNetwork)
+        {
+            Debug.Assert(addressFamily == AddressFamily.InterNetwork || addressFamily == AddressFamily.InterNetworkV6);
+            return addressFamily == AddressFamily.InterNetwork ? s_ipv4Socket : s_ipv6Socket;
+        }
+
+        [Fact]
+        public void BeginAccept_NegativeReceiveSize_Throws_ArgumentOutOfRange()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => GetSocket().BeginAccept(null, -1, TheAsyncCallback, null));
```

Comment:
Do any of these argument exceptions include the name of the parameter?  If so, we could add that name as the first argument to Throws, and it'll verify the parameter name as well.